python/packages/core/agent_framework/_workflows/_agent_executor.py

```diff
@@ -0,0 +1,201 @@
+        *,
+        agent_thread: AgentThread | None = None,
+        output_response: bool = False,
+        id: str | None = None,
+    ):
+        """Initialize the executor with a unique identifier.
+
+        Args:
+            agent: The agent to be wrapped by this executor.
+            agent_thread: The thread to use for running the agent. If None, a new thread will be created.
+            output_response: Whether to yield an AgentRunResponse as a workflow output when the agent completes.
+            id: A unique identifier for the executor. If None, the agent's name will be used if available.
+        """
+        # Prefer provided id; else use agent.name if present; else generate deterministic prefix
+        if id is not None:
+            exec_id = id
+        else:
+            agent_name = agent.name
+            if agent_name:
+                exec_id = str(agent_name)
+            else:
+                raise ValueError("Agent must have a name or an explicit id must be provided.")
+        super().__init__(exec_id)
+        self._agent = agent
+        self._agent_thread = agent_thread or self._agent.get_new_thread()
+        self._output_response = output_response
+        self._cache: list[ChatMessage] = []
+
+    @property
+    def workflow_output_types(self) -> list[type[Any]]:
+        # Override to declare AgentRunResponse as a possible output type only if enabled.
+        if self._output_response:
+            return [AgentRunResponse]
+        return []
+
+    async def _run_agent_and_emit(self, ctx: WorkflowContext[AgentExecutorResponse, AgentRunResponse]) -> None:
+        """Execute the underlying agent, emit events, and enqueue response.
+
+        Checks ctx.is_streaming() to determine whether to emit incremental AgentRunUpdateEvent
+        events (streaming mode) or a single AgentRunEvent (non-streaming mode).
+        """
+        if ctx.is_streaming():
+            # Streaming mode: emit incremental updates
+            updates: list[AgentRunResponseUpdate] = []
+            async for update in self._agent.run_stream(
+                self._cache,
+                thread=self._agent_thread,
+            ):
+                # Skip empty updates (no textual or structural content)
+                if not update:
+                    continue
+                contents = getattr(update, "contents", None)
+                text_val = getattr(update, "text", "")
+                has_text_content = False
+                if contents:
+                    for c in contents:
+                        if getattr(c, "text", None):
+                            has_text_content = True
+                            break
+                if not (text_val or has_text_content):
+                    continue
+                updates.append(update)
+                await ctx.add_event(AgentRunUpdateEvent(self.id, update))
+            response = AgentRunResponse.from_agent_run_response_updates(updates)
```

Comment:
You might want to add a potential structured output to this call, so that we get that consistent across streaming and nonstreaming

Reply:
How do we know the structured output format from here.